### ISSUE TITLE
about control frequency

### ISSUE THREAD

**Issue by tianhaowuhz** [justagist/franka_ros_interface#34]
Hi, Im recently trying to use this api, and Im wondering what's the highest control frequency can this api reach, I have tested like moving continuously, and it's about 1s for each action, is there is a way to have higher frequency?

**Comment by tianhaowuhz** [comment]
I am so sorry, there seems no notification, i didn't notice you already respond to me. I am trying to use exec cmd in python wrapper, is these the command you mentioned that can reach the rate of 250 commands per second?

**Comment by tianhaowuhz** [comment]
also ,I am wondering which api in python should I use if I want to make panda move more smooth like when track a marker?

**Comment by justagist** [comment]
> I am so sorry, there seems no notification, i didn't notice you already respond to me. I am trying to use exec cmd in python wrapper, is these the command you mentioned that can reach the rate of 250 commands per second?

Yes, I am able to attain over 250Hz with the python commands. Note that this requires you to send commands in a loop that is running independently at at least that rate. Simple example [here](https://github.com/justagist/franka_ros_interface/blob/bf4660f33d16a51925b362ee0690b5565f59e0c3/franka_ros_controllers/scripts/test_controller.py).

> also ,I am wondering which api in python should I use if I want to make panda move more smooth like when track a marker?

You can still use the python code to test provide smooth motion; you have to make sure that the provided trajectory is very smooth and continuous, especially if you use position control (see #28). But motion should be much smoother if you use torque or velocity control. If you want the robot tracking a marker, I would suggest writing a task-space position/impedance controller using the torque command interface (a vaguely similar example is the task-space demo in [panda_simulator](https://github.com/justagist/panda_simulator)).
